tests/functional/object/mcg/test_s3_vector.py

```
@@ -0,0 +1,206 @@
+logger = logging.getLogger(__name__)
+
+
+@mcg
+@red_squad
+class TestS3VectorWithFactory(MCGTest):
+    """
+    Test S3 Vector functionality against noobaa
+    """
+
+    @tier2
+    def test_s3_vector_operations_with_factory(self, vector_bucket_factory, mcg_obj):
```

Ruff (0.15.13)

[warning] Too many statements (62 > 50)

(PLR0915)